src/app/modules/submission/submission.service.ts

```diff
@@ -1403,5 +1420,16 @@ export const triggerGuarddutyScanThenDownloadCleanFileChain = <
           answer: response.filename,
         })),
       )
+      .mapErr((error) => {
```

Comment:
Exception wrapping

Reply:
Not too sure about this since each of the `DownloadCleanFileFailedError` (the error that's being wrapped) would have also done a `span.setTag`.

I believe the wrapping would make sense to wrap primitive errors rather than ApplicationError.

Reply:
Looked around, couldn't find anything better. Ok with solution. But not something we want to do more often (overriding error thrown)

Reply:
I actually tried to directly throw the new errors from the source eg, from `downloadCleanFile`. The problem was that this function is shared with `triggerVirusScannerAndScan` and `triggerGuardDutyVirusScannerAndScan` and i will either need to: 
- duplicate the function to create a guard duty version 
- pass in a flag `isGuardDuty` to conditionally throw the error based on this flag

These 2 approaches don't seem too good either - and require more invasive changes. 

Agreed that this is not something we want to do regularly, but due to the nature of the virus scanner code, think this is the least invasive way to add this functionality.